sidechain/top-pool-executor/src/call_operator.rs

```diff
@@ -17,107 +17,47 @@
 
```

Comment:
There have been some big changes here, and I'm only half way through. I wanted to get your approval first.

I started to remove the stf interaction part completely from the top-pool-executor. In my eyes these two do not belong into one "executor" - we have two anyway, why not use them separately?

So this call_oparator now only interacts with the top pool.
My vision: Also adapt the getter_opator accordingly. Does this go against your vision completely?

Reply:
It certainly changes my initial design motivation for this component. The idea was in the `TopPoolExecutor` to abstract the execution of top-pool operations, which uses a combination of the `StfExecutor` (to do all the necessary state changes) and the top-pool (to retrieve, remove). So the client/caller of a `TopPoolExecutor` does not need to be concerned about any state changes or top pool interactions (and thus does not need to know about the `StfExecutor` or the top-pool).

With these changes now, we've basically removed this layer of abstraction. The `TopPoolExecutor` is basically just a top-pool facade and all the client have to use `StfExecutor` to do the state changes. So I believe we need to re-think our design. As it stands right now, it's a bit of a mess.
I'll draw some diagrams to better illustrate my point and append them here.

Reply:
So @haerdib had a discussion in person and we came to the conclusion:
* For this PR we'll keep the design as proposed here
* In a future PR/issue, we remove the `call_operator`, and directly use the `RpcAuthor`
  * Need to update/refactor the `RpcAuthor` interface a bit to suit our needs
  * The remaining code in `TopPoolExecutor` will only be related to executing trusted getters, so we should probably re-name the crate to something getter related.

Reply:
On a second thought, I think it makes sense to fix it in this PR right away, it should not be that much of a conceptual change.

@clangenb do you agree on our take to remove the TopPoolCallOperator / Executor completely and rename the TopPoolExecutor to TrustedGetterExecutor?

Reply:
I kind of dislike trusted getter 'executor' because it only gets something, it doesn't really execute. I am struggling to find an alternative now though.

Reply:
Well I would argue that it is executing a getter. The getter is a bit involved - it does have good amount of logic surrounding it, and it's 'executing' that getter logic.

Reply:
Resolving this discussion in favor of: https://github.com/integritee-network/worker/issues/547